x-pack/plugins/ml/public/embeddables/anomaly_swimlane/anomaly_swimlane_setup_flyout.tsx

```diff
@@ -27,7 +28,7 @@ export async function resolveAnomalySwimlaneUserInput(
 
   return new Promise(async (resolve, reject) => {
     try {
-      const { jobIds } = await resolveJobSelection(coreStart, input?.jobIds);
+      const { jobIds } = await resolveJobSelection(coreStart, isServerless, input?.jobIds);
```

Comment:
Did you consider removing all node-related info server-side? It seems like it should simplify the UI code.

Reply:
We need the node information client side because we use it to determine whether a job is awaiting node assignment and can then inform the user.
This is the case for AD and DFA jobs.

Having to pass the `isServerless` flag about for node filtering is only really annoying for the AD jobs list which is still written in js and so it needs to be added as a prop to few components.
When this gets converted to ts we can remove the `isServerless` props on these components and just get the flag from the context.
It's worth noting that most of the usage of the `isServerless` flag is not for removing node information. It is used in `getMlGlobalServices` which creates the data for our global context. `getMlGlobalServices` is reused to wrap context around embeddables and some flyouts and so the flag needs to be passed to the functions which register this code.